server/text_generation_server/models/custom_modeling/flash_rw_modeling.py

```diff
@@ -312,21 +311,17 @@ def forward(
         # Inplace rotary
         self.rotary_emb(query, torch.select(kv, dim=2, index=0), cos, sin)
 
-        reshape_and_cache(
-            kv[:, :, 0].contiguous(),
-            kv[:, :, 1].contiguous(),
-            kv_cache[0],
-            kv_cache[1],
-            slots,
+        kv_cache.store(
+            key=kv[:, :, 0].contiguous(), value=kv[:, :, 1].contiguous(), slots=slots
```

Comment:
not important, but its strange we need the `.contiguous()` calls here

Reply:
Yeah, I wasn't sure if it was needed, these seems to be key/value striding in other places that is non-contiguous, but I also didn't want to touch it.